demo/random.cpp

```diff
@@ -1,134 +1,134 @@
-        s = s * 2891336453 + 1;
-        buffer[i] = s;
-    }
-
-    // Initialize pointers into the buffer
-    p1 = 0;
-    p2 = 10;
-}
-
-unsigned Random::rotl(unsigned n, unsigned r)
-{
-    return (n << r) | (n >> (32 - r));
-}
-
-unsigned Random::rotr(unsigned n, unsigned r)
-{
-    return (n >> r) | (n << (32 - r));
-}
-
-unsigned Random::randomBits()
-{
-    unsigned result;
-
-    // Rotate the buffer and store it back to itself
-    result = buffer[p1] = rotl(buffer[p2], 13) + rotl(buffer[p1], 9);
-
-    // Rotate pointers
-    if (--p1 < 0)
-        p1 = 16;
-    if (--p2 < 0)
-        p2 = 16;
-
-    // Return result
-    return result;
-}
-
-#ifdef SINGLE_PRECISION
-real Random::randomReal()
-{
-    // Get the random number
-    unsigned bits = randomBits();
-
-    // Set up a reinterpret structure for manipulation
-    union {
-        real value;
-        unsigned word;
-    } convert;
-
-    // Now assign the bits to the word. This works by fixing the ieee
-    // sign and exponent bits (so that the size of the result is 1-2)
-    // and using the bits to create the fraction part of the float.
-    convert.word = (bits >> 9) | 0x3f800000;
-
-    // And return the value
-    return convert.value - 1.0f;
-}
-#else
-real Random::randomReal()
-{
-    // Get the random number
-    unsigned bits = randomBits();
-
-    // Set up a reinterpret structure for manipulation
-    union {
-        real value;
-        unsigned words[2];
-    } convert;
-
-    // Now assign the bits to the words. This works by fixing the ieee
-    // sign and exponent bits (so that the size of the result is 1-2)
-    // and using the bits to create the fraction part of the float. Note
-    // that bits are used more than once in this process.
-    convert.words[0] = bits << 20; // Fill in the top 16 bits
-    convert.words[1] = (bits >> 12) | 0x3FF00000; // And the bottom 20
-
-    // And return the value
-    return convert.value - 1.0;
-}
-#endif
-
-real Random::randomReal(real min, real max)
-{
-    return randomReal() * (max - min) + min;
-}
-
-real Random::randomReal(real scale) { return randomReal() * scale; }
-
-unsigned Random::randomInt(unsigned max) { return randomBits() % max; }
-
-real Random::randomBinomial(real scale)
-{
-    return (randomReal() - randomReal()) * scale;
-}
-
-Vector3 Random::randomVector(real scale)
-{
-    return Vector3(randomBinomial(scale), randomBinomial(scale),
-        randomBinomial(scale));
-}
-
-Vector3 Random::randomXZVector(real scale)
-{
-    return Vector3(randomBinomial(scale), 0, randomBinomial(scale));
-}
-
-Vector3 Random::randomVector(const Vector3& scale)
-{
-    return Vector3(randomBinomial(scale.x), randomBinomial(scale.y),
-        randomBinomial(scale.z));
-}
-
-Vector3 Random::randomVector(const Vector3& min, const Vector3& max)
-{
-    return Vector3(randomReal(min.x, max.x), randomReal(min.y, max.y),
-        randomReal(min.z, max.z));
-}
+#include "demo/random.hpp"
```

Comment:
If you are not the author of this file, please add license information and/or reference to original (if there is a link - link it, if there is a book name/isbn - mention it)